core/src/main/java/org/apache/druid/math/expr/Function.java

```diff
@@ -1453,4 +1512,428 @@ public ExprEval apply(List<Expr> args, Expr.ObjectBinding bindings)
     }
   }
 
+  class StringToArrayFunction implements Function
+  {
+    @Override
+    public String name()
+    {
+      return "string_to_array";
+    }
+
+    void validateArguments(List<Expr> args)
+    {
+      if (args.size() != 2) {
+        throw new IAE("Function[%s] needs 2 argument", name());
+      }
+    }
+
+    @Override
+    public ExprEval apply(List<Expr> args, Expr.ObjectBinding bindings)
+    {
+      validateArguments(args);
+
+      final ExprEval expr = args.get(0).eval(bindings);
+      final String arrayString = expr.asString();
+      if (arrayString == null) {
+        return ExprEval.of(null);
+      }
+
+      final String split = args.get(1).eval(bindings).asString();
+      return ExprEval.ofStringArray(arrayString.split(split != null ? split : ""));
+    }
+
+    @Override
+    public Set<Expr> getScalarInputs(List<Expr> args)
+    {
+      validateArguments(args);
+      return ImmutableSet.copyOf(args);
+    }
+  }
+
+  /**
+   * Function that operates on array typed operands
+   */
+  interface ArrayFunction extends Function
+  {
+    default void validateArguments(List<Expr> args)
```

Comment:
nit: this are other default methods on the interface are never used as all implementations seem to be overriding all of those, if we remove the default impls then this method probably doesn't need to exist in the interface as none of the ArrayFunction users call it.

Reply:
It is used in one place, `ArrayLengthFunction`, but that's still lame. Will try to clean this area up a bit.